gobblin-metrics/src/main/java/gobblin/metrics/example/ReporterExampleBase.java

```diff
@@ -0,0 +1,183 @@
+/* (c) 2014 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.metrics.example;
+
+import java.io.IOException;
+import java.util.Random;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Timer;
+
+import gobblin.metrics.ContextAwareScheduledReporter;
+import gobblin.metrics.MetricContext;
+import gobblin.metrics.Tag;
+
+
+/**
+ * A base class for those that exemplifies the usage of {@link MetricContext} and a given
+ * {@link ContextAwareScheduledReporter}.
+ *
+ * <p>
+ *   This class simulating a record processing job that spawns a configurable number of tasks
+ *   running in a thread pool for processing incoming data records in parallel. Each task will
+ *   simulate processing the same configurable number of records. The job creates a job-level
+ *   {@link MetricContext}, from which a child {@link MetricContext} for each task is created.
+ *   Each task uses four metrics in its {@link MetricContext} to keep track of the total number
+ *   of records processed, record processing rate, record size distribution, and record processing
+ *   times. Since the job-level {@link MetricContext} is the parent of the {@link MetricContext}s
+ *   of the tasks, updates to a metric of a task will be automatically applied to the metric with
+ *   the same name in the job-level {@link MetricContext}. The job and task {@link MetricContext}s
+ *   use the same given {@link ContextAwareScheduledReporter} to report metrics.
+ * </p>
+ *
+ * @author ynli
+ */
+public class ReporterExampleBase {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger("ReporterExample");
```

Comment:
Can we do getLogger(ReporterExampleBase.class) instead (this plays nicer with logger configurations).